src/foam/u2/ActionView.js

```diff
@@ -344,11 +344,11 @@ foam.CLASS({
       if ( this.action ) {
         this.attrs({ name: this.action.name });
 
-        if ( this.action.isAvailable ) {
+        if ( this.action.isAvailable || this.action.permissionConfig != null ) {
```

Comment:
This seems overly complex.  We don't need to update all the views to understand permissions, all we need to do us ensure that the isAvailable and isEnabled slot's created by the action _also_ take permissions into account if necessary.

I think we should just put enabledPermissions and availablePermissions as properties on Action, and then expand the createIsEnabled and createIsAvailable code to create slots which also check the permissions.

As a result the interface to an Action never changes and _all_ the views which currently check isEnabled/isAvailable will pick up permission support automatically.

Reply:
> This seems overly complex. We don't need to update all the views to understand permissions, all we need to do us ensure that the isAvailable and isEnabled slot's created by the action _also_ take permissions into account if necessary.

They already do. `createIsEnabled$` and `createIsAvailable$` already take permissions into account. The only reason we have these `if`s in ActionView is as a tiny optimization to avoid setting up the slots if they won't be doing anything anyway. ActionView would work just as well without these checks and other ActionViews will as well.

> I think we should just put enabledPermissions and availablePermissions as properties on Action...

I can change the interface to let you set those instead of providing an `ActionPermissionConfiguration` object if you think that's a better interface.